pkg/util/test/util.go

```diff
@@ -217,3 +225,190 @@ func ActivateCRDWhenCreated(fakeClient *fakeextclient.Clientset, crdName string)
 		return false, nil
 	})
 }
+
+// UninstallStorageCluster uninstall and wipe storagecluster from k8s
+func UninstallStorageCluster(cluster *corev1alpha1.StorageCluster) error {
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if cluster.Spec.DeleteStrategy == nil ||
+		(cluster.Spec.DeleteStrategy.Type != corev1alpha1.UninstallAndWipeStorageClusterStrategyType &&
+			cluster.Spec.DeleteStrategy.Type != corev1alpha1.UninstallStorageClusterStrategyType) {
+		cluster.Spec.DeleteStrategy = &corev1alpha1.StorageClusterDeleteStrategy{
+			Type: corev1alpha1.UninstallAndWipeStorageClusterStrategyType,
+		}
+		if _, err = k8s.Instance().UpdateStorageCluster(cluster); err != nil {
+			return err
+		}
+	}
+
+	return k8s.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
+}
+
+// ValidateStorageCluster validates a StorageCluster spec
+func ValidateStorageCluster(
+	cluster *corev1alpha1.StorageCluster,
+	timeout, interval time.Duration,
+) error {
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if err = validateStorageClusterPods(cluster, timeout, interval); err != nil {
+		return err
+	}
+
+	svc, err := k8s.Instance().GetService("portworx-service", cluster.Namespace)
+	pxEndpoint := ""
+	if err != nil {
+		return err
+	}
+
+	servicePort := int32(9020)
+	for _, port := range svc.Spec.Ports {
+		if strings.Contains(port.Name, "sdk") {
```

Comment:
The port is called px-sdk. We can directly compare the name instead of doing Contains.
Should error out if there is no px-sdk port in the service.